diff --git a/build_lib.py b/build_lib.py
index 91a1c1d9..2ee38be0 100644
--- a/build_lib.py
+++ b/build_lib.py
@@ -278,6 +278,9 @@ def main(argv: list[str] | None = None) -> int:
                 print("Warp build error: Could not find MSVC compiler")
                 return 1

+    else:
+        args.host_compiler = build_dll.find_host_compiler()
+
     try:
         # Generate warp/native/export.h
         generate_exports_header_file(base_path)
diff --git a/warp/build_dll.py b/warp/build_dll.py
index a2fa5d74..9638b5b9 100644
--- a/warp/build_dll.py
+++ b/warp/build_dll.py
@@ -120,7 +120,7 @@ def find_host_compiler():

     else:
         # try and find g++
-        return run_cmd("which g++").decode()
+        return run_cmd("echo $CXX").decode().strip() if run_cmd("echo $CXX") != b"\xA0" else run_cmd("which g++").decode().strip()


 def get_cuda_toolkit_version(cuda_home) -> tuple[int, int]:
@@ -387,7 +387,7 @@ def build_dll_for_arch(args, dll_path, cpp_paths, cu_path, arch, libs: list[str]
     else:
         # Unix compilation
         cuda_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else "nvcc"
-        cpp_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else "g++"
+        cpp_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else args.host_compiler

         cpp_includes = f' -I"{warp_home_path.parent}/external/llvm-project/out/install/{mode}-{arch}/include"'
         cpp_includes += f' -I"{warp_home_path.parent}/_build/host-deps/llvm-project/release-{arch}/include"'
