diff --git a/build_llvm.py b/build_llvm.py
--- a/build_llvm.py
+++ b/build_llvm.py
@@ -67,11 +67,6 @@ def fetch_prebuilt_libraries(arch):
 def build_from_source_for_arch(args, arch, llvm_source):
     # Check out the LLVM project Git repository, unless it already exists
     if not os.path.exists(llvm_source):
-        # Install dependencies
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "gitpython"])
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "cmake"])
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "ninja"])
-
         from git import Repo

         repo_url = "https://github.com/llvm/llvm-project.git"
